### PR TITLE
fix: Accidentally publishing useless files

### LINKED ISSUE #16916: [Bug]: Path Error Showing
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

@babel/eslint-parser

### Input code

/* This file is automatically generated by scripts/generators/tsconfig.js */
{
  "extends": [
    "../../tsconfig.base.json",
    "../../tsconfig.paths.json"
  ],
  "include": [
    "../../eslint/babel-eslint-parser/src/**/*.cts",
    "../../lib/globals.d.ts",
    "../../scripts/repo-utils/*.d.ts"
  ],
  "references": [
    {
      "path": "../../packages/babel-core"
    }
  ]
}

### Configuration file name

_No response_

### Configuration

Here is my eslint.config.js
import js from '@eslint/js';
import globals from 'globals';
import babelParser from '@babel/eslint-parser';
import prettierPlugin from 'eslint-plugin-prettier';

export default [
  {
    languageOptions: {
      ecmaVersion: 'latest',
      sourceType: 'module',
      parser: babelParser,
      parserOptions: {
        requireConfigFile: false,
        babelOptions: {
          babelrc: false,
          configFile: false,
          presets: ['@babel/preset-env'],
        },
      },
      globals: {
        ...globals.browser,
        ...globals.node,
        process: true,
        __dirname: true,
      },
    },
  },
  {
    files: ['**/*.js'],
    plugins: {
      prettier: prettierPlugin,
    },
    ...js.configs.recommended,
    rules: {
      'prettier/prettier': [
        'error',
        {
          singleQuote: true,
          semi: true,
          tabWidth: 2,
          trailingComma: 'all',
          printWidth: 100,
          bracketSpacing: true,
          arrowParens: 'always',
        },
      ],
      'no-unused-vars': [
        'warn',
        {
          argsIgnorePattern: '^_',
          varsIgnorePattern: '^_',
          caughtErrorsIgnorePattern: '^_',
        },
      ],
      'no-undef': 'error',
      'no-var': 'error',
      'prefer-const': 'error',
      'no-use-before-define': 'error',
      'arrow-body-style': ['error', 'as-needed'],
      'prefer-arrow-callback': 'error',
      'prefer-template': 'error',
      'object-shorthand': 'error',
      'prefer-destructuring': 'warn',
      'no-console': ['warn', { allow: ['warn', 'error', 'info'] }],
      eqeqeq: ['error', 'always'],
      curly: ['error', 'all'],
      'no-throw-literal': 'error',
      'require-await': 'error',
      'no-return-await': 'error',
      'max-len': [
        'error',
        {
          code: 100,
          ignoreComments: true,
          ignoreStrings: true,
          ignoreTemplateLiterals: true,
          ignoreRegExpLiterals: true,
        },
      ],
    },
  },
  {
    files: ['src/**/*.js'],
    rules: {
      'no-console': ['error', { allow: ['error', 'warn'] }],
      'handle-callback-err': 'error',
      'no-buffer-constructor': 'error',
      'no-mixed-requires': 'error',
      'no-path-concat': 'error',
      'consistent-return': 'error',
      'no-param-reassign': 'error',
    },
  },
  {
    files: ['tests/**/*.js', '**/*.test.js', '**/*.spec.js'],
    rules: {
      'no-unused-vars': 'off',
      'no-console': 'off',
      'max-len': 'off',
      'require-await': 'off',
      'max-lines-per-function': 'off',
    },
  },
  {
    files: ['config.js', '*.config.js'],
    rules: {
      'no-console': 'off',
    },
  },
];

### Current and expected behavior

File '/home/murshid/Documents/GitHub/quote_app/node_modules/packages/babel-core' not found.ts
Path to referenced tsconfig or to folder containing tsconfig.
I got this error node_modules/@bable/eslint-parser/tsconfig.json
Screenshot from 2024-10-19 12-14-49
Why this error showing ? i reinstalled all the packages 

### Environment

{
  "name": "quote_app",
  "version": "1.0.0",
  "main": "server.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "nodemon server.js"
  },
  "author": "",
  "license": "ISC",
  "description": "",
  "devDependencies": {
    "@babel/core": "^7.25.8",
    "@babel/eslint-parser": "^7.25.8",
    "@babel/preset-env": "^7.25.8",
    "@eslint/js": "^9.13.0",
    "eslint": "^9.13.0",
    "eslint-plugin-import": "^2.31.0",
    "eslint-plugin-prettier": "^5.2.1",
    "prettier": "^3.3.3"
  },
  "dependencies": {
    "globals": "^15.11.0"
  }
}

### Possible solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #16916 
| Patch: Bug Fix?          |√
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

We actually publish `tsconfig.json` and even `tsconfig.tsbuildinfo` for all packages.🤦‍♂️

### CODE REVIEW

File: packages/babel-standalone/package.json [hunk 1]
```diff
@@ -4,10 +4,10 @@
...
   "main": "./babel.js",
   "files": [
-    "babel.js",
-    "babel.js.map",
-    "babel.min.js",
-    "babel.min.js.map"
+    "./babel.js",
+    "./babel.js.map",
+    "./babel.min.js",
+    "./babel.min.js.map"
```
Comment (liuxingbaoyu) [inline packages/babel-standalone/package.json:10]: image

**Review by nicolo-ribaudo — APPROVED** [review]
Do you have any idea of why we started accidentally publishing them?

**Comment by liuxingbaoyu** [comment]
> Do you have any idea of why we started accidentally publishing them?

https://github.com/babel/babel/pull/16430
It seems that starting from this PR, each package has its own `tsconfig.json`.

File: codemods/babel-plugin-codemod-object-assign-to-object-spread/.npmignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 src
 test
-*.log
```
Comment (JLHwung) [inline codemods/babel-plugin-codemod-object-assign-to-object-spread/.npmignore:3]: I suggest we keep the `*.log` here. In circumstances where we have to manually publish npm packages and the first publish attempt is failed, npm will generate a debug log `npm-debug.log`, if we forget to remove that during the second publish attempt, this log will be unfortunately published.

**Review by JLHwung — APPROVED** [review]

**Pull request merged**